lib/src/client.dart

```diff
@@ -1011,6 +1011,20 @@ class Client extends MatrixApi {
     _archivedRooms.add(ArchivedRoom(room: archivedRoom, timeline: timeline));
   }
 
+  bool _isServerConfigCached = false;
+  ServerConfig? _cachedServerConfig;
```

Comment:
Do we need the bool? 

Also caching indefinitely sounds wrong, probably add a timeout of say 15 min (random number)

Reply:
I'm caching till we call `init` again. I think that's when the homeserver changes. Do you think we should cache based on timeout?

Reply:
getConfig values could change and having it reflect without restart the client would be nice :D

Reply:
I can see the [go matrix sdk's bridge part](https://github.com/mautrix/go/blob/685d5f71da580e63c0f524e103abe83452d54d11/bridge/bridge.go#L637) and [python sdk's bridge part](https://github.com/mautrix/python/blob/8d39c733d7ad2d1a11b70c255babfad6941faf3a/mautrix/bridge/matrix.py#L259) caching it only once in the `start` call that comes after `init`. 

The both have a default value for the "m.upload.size" too - `50 * 1024 * 1024`.

Reply:
You could use an AsyncCache with a cache duration of one hour probably? https://pub.dev/documentation/async/latest/async/AsyncCache-class.html